### PR TITLE
F port kinematic sim from kinetic

### PR DESCRIPTION
**Pull request by toliver**
## Proposed changes

Changes ported from kinetic-devel.

## Types of changes

What types of changes does your code introduce?

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

Put an x in the boxes that apply. You can also fill these out after creating the PR. This is a reminder of what we should look for before merging this code. I have:

- [ ] Read and follow the [contributing guidelines](https://github.com/shadow-robot/sr_documentation/blob/master/CONTRIBUTING.md).
- [ ] Checked that all tests pass with my changes
- [ ] Added tests (automatic or manual) that prove the fix is effective or that the feature works
- [ ] Added necessary documentation (if appropriate)
- [ ] Added the corresponding license to each file and add the current year of any one that you modified.
- [ ] Tested on real hardware (if appropriate)

### CODE REVIEW

File: sr_utilities_common/launch/simple_kinematic_simulator.launch [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+                                       rh_RFJ1, rh_RFJ2, rh_RFJ3, rh_RFJ4,
+                                       rh_LFJ1, rh_LFJ2, rh_LFJ3, rh_LFJ4, rh_LFJ5,
+                                       rh_THJ1, rh_THJ2, rh_THJ3, rh_THJ4, rh_THJ5,
+                                       rh_WRJ1, rh_WRJ2]"/>
+  <!-- If a robot_side "right" is specified, any joint in the trajecotry cmd that has a prefix lh or la will be 
+      republished as rh and ra
+       If a robot side "left" is specified, any joint in the trajecotry cmd that has a prefix rh or ra will be 
+      republished as lh and la
+       If robot_side is empty "", then joints will be published without name changes -->
+  <arg name="robot_side" default="right"/>
```
Comment (mikramarc) [inline sr_utilities_common/launch/simple_kinematic_simulator.launch:17]: we started using this convention when dealing with side and prefixes, could you do that too?
https://github.com/shadow-robot/sr_teleop_vive_polhemus_internal/blob/melodic-devel/sr_teleop_vive_polhemus/launch/polhemus.launch#L2

File: sr_utilities_common/launch/simple_kinematic_simulator.launch [hunk 2]
```diff
@@ -0,0 +1,38 @@
+<launch>
+  <arg name="robot_description" default="'$(find sr_description)/robots/shadowhand_motor_plus.urdf.xacro'"/>
+  <arg name="remapped_joint_states_topic_name" default="remapped_joint_states"/>
+  <arg name="trajectory_cmd_topic_name" default="/rh_trajectory_controller/command"/>
+  <arg name="joints_to_move" default="[rh_FFJ1, rh_FFJ2, rh_FFJ3, rh_FFJ4,
```
Comment (mikramarc) [inline sr_utilities_common/launch/simple_kinematic_simulator.launch:6]: prefixes will be affected by the side below too

File: sr_utilities_common/launch/simple_kinematic_simulator.launch [hunk 3]
```diff
@@ -0,0 +1,38 @@
+<launch>
+  <arg name="robot_description" default="'$(find sr_description)/robots/shadowhand_motor_plus.urdf.xacro'"/>
+  <arg name="remapped_joint_states_topic_name" default="remapped_joint_states"/>
+  <arg name="trajectory_cmd_topic_name" default="/rh_trajectory_controller/command"/>
```
Comment (mikramarc) [inline sr_utilities_common/launch/simple_kinematic_simulator.launch:4]: prefixes will be affected by the side below

File: sr_utilities_common/launch/simple_kinematic_simulator.launch [hunk 4]
```diff
@@ -0,0 +1,38 @@
+<launch>
+  <arg name="robot_description" default="'$(find sr_description)/robots/shadowhand_motor_plus.urdf.xacro'"/>
```
Comment (mikramarc) [inline sr_utilities_common/launch/simple_kinematic_simulator.launch]: that will change depending on the side, right? Can we parametrize this?

File: sr_utilities_common/scripts/sr_utilities_common/republish_trajectory_as_joint_states.py [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+import rospy
+from sensor_msgs.msg import JointState
+from trajectory_msgs.msg import JointTrajectory, JointTrajectoryPoint
+
+
+class RePubTrajectoryAsJointStates(object):
+    def __init__(self,
+                 joints_to_move,
+                 robot_side="",
```
Comment (mikramarc) [inline sr_utilities_common/scripts/sr_utilities_common/republish_trajectory_as_joint_states.py]: should we set a default to right sice we're setting a defautl for trajectory_cmd_topic as well?

File: sr_utilities_common/scripts/sr_utilities_common/republish_trajectory_as_joint_states.py [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+                                            JointTrajectory,
+                                            self._traj_cb,
+                                            tcp_nodelay=True)
+        self._joint_states_pub = rospy.Publisher(joint_states_pub_topic,
+                                                 JointState,
+                                                 queue_size=10)
+        self._joints_to_move = joints_to_move
+        self._right_hand_prefix = "rh_"
+        self._left_hand_prefix = "lh_"
+        self._right_arm_prefix = "rh_"
```
Comment (mikramarc) [inline sr_utilities_common/scripts/sr_utilities_common/republish_trajectory_as_joint_states.py]: isnt right arm prefix "ra", not "rh"?

**Review by mikramarc — DISMISSED** [review]

**Comment by toliver** [comment]
oops, @mikramarc I've just realized you merged this into melodic-devel, but the PR remained open, is that correct?
If that's the case I will take the chance and add some improvements to the PR.

**Review by biofotis — APPROVED** [review]

File: sr_utilities_common/launch/bimanual_simple_kinematic_simulator.launch [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+  <arg name="robot_description" default="$(find sr_multi_description)/urdf/bimanual_srhand_$(arg robot_model).urdf.xacro"/>
+
+  <param name="robot_description" command="xacro --inorder $(arg robot_description)"/>
+
+  <!-- Right hand -->
+  <include file="$(find sr_utilities_common)/launch/single_robot_position_republisher.launch">
+    <arg name="node_name" value="rh_trajectory_republisher"/>
+    <arg name="remapped_joint_states_topic_name" value="rh_remapped_joint_states"/>
+    <arg name="trajectory_cmd_topic_name" value="/rh_trajectory_controller/command"/>
+    <arg name="joints_to_move" value="[]"/>s
```
Comment (mikramarc) [inline sr_utilities_common/launch/bimanual_simple_kinematic_simulator.launch]: there is an 's' sign at the end of this line and a few others

File: sr_utilities_common/launch/right_hand_simple_kinematic_simulator.launch [hunk 8]
```diff
@@ -0,0 +1,25 @@
+<launch>
+  <arg name="robot_description" default="'$(find sr_description)/robots/shadowhand_motor_plus.urdf.xacro'"/>
+
+  <param name="robot_description" command="xacro --inorder $(arg robot_description)"/>
+
+  <include file="$(find sr_utilities_common)/launch/single_robot_position_republisher.launch">
+    <arg name="node_name" value="rh_trajectory_republisher"/>
+    <arg name="remapped_joint_states_topic_name" value="rh_remapped_joint_states"/>
+    <arg name="trajectory_cmd_topic_name" value="/rh_trajectory_controller/command"/>
+    <arg name="joints_to_move" value="[]"/>s
```
Comment (mikramarc) [inline sr_utilities_common/launch/right_hand_simple_kinematic_simulator.launch]: and here

**Review by mikramarc — CHANGES_REQUESTED** [review]

File: sr_utilities_common/launch/bimanual_simple_kinematic_simulator.launch (hunk 7 above)
Comment (biofotis) [reply sr_utilities_common/launch/bimanual_simple_kinematic_simulator.launch]: fixed

**Review by mikramarc — APPROVED** [review]

**Pull request merged**